website_mass_mailing_name/static/src/js/website_mass_mailing_name.js

```diff
@@ -0,0 +1,60 @@
+/* Copyright 2016-2017 Jairo Llopis <jairo.llopis@tecnativa.com>
+ * License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl). */
+
+odoo.define("website_mass_mailing_name.subscribe", function (require) {
+    "use strict";
+    require("mass_mailing.website_integration");
+    var animation = require("website.content.snippets.animation");
+
+    animation.registry.subscribe.include({
+        start: function(editable_mode) {
+            this.$email = this.$target.find(".js_subscribe_email");
+            this.$name = this.$target.find(".js_subscribe_name");
+            // Thanks upstream for your @$&#?!! inheritance-ready code.
+            // Injecting ajax events to modify behavior of snippet.
+            if (this.$name) {
+                $(document).ajaxSend($.proxy(this.on_ajax_send, this));
+            }
+            return this._super(editable_mode);
+        },
+
+        _onClick: function() {
+            var email_error = !this.$email.val().match(/.+@.+/),
```

Comment:
Here uses regex only to "check" that string have an @... perhaps can be better use "search" instead of "match".

Reply:
Same as the comment above. Would be better move forward instead spend more time?